src/main/scala/com/kinja/amqp/UnconfirmedMessageRepeater.scala

```diff
@@ -0,0 +1,72 @@
+package com.kinja.amqp
+
+import com.kinja.amqp.model.Message
+import com.kinja.amqp.model.MessageConfirmation
+
+import org.slf4j.{ Logger => Slf4jLogger }
+
+import scala.util.Failure
+import scala.util.Success
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object UnconfirmedMessageRepeater {
+
+	/**
+	 * Loads the messages from the store.
+	 * For already confirmed messages, deletes the message and the confirmation.
+	 * For unconfirmed messages, tries to republish and if succeeds, deletes old message and confirmation.
+	 */
+	def resendUnconfirmedMessages(messageStore: MessageStore, olderThan: Long, producer: AmqpProducer, logger: Slf4jLogger): Unit = {
```

Comment:
Pass in producers that is a map of [exchangeName, Producer] and iterate through all the exchanges. This also means that we have to store the exchange name with the message and add it to ther relevant queries.